docs/source/security/cors.mdx

```diff
@@ -0,0 +1,108 @@
+---
+title: Configuring CORS
+description: Control access to your server's resources
+---
+
+[Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) (CORS) is an HTTP-header based protocol that enables a server to dictate which origins have access to its resources. Put another way, your server can specify which client's scripts can read the contents of your server's response.
+
+## Why use CORS?
+
+Most developers know about CORS because they run into the all-too-common [CORS error](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors). CORS errors usually occur when you set up an API call or try to get your separately hosted server and client to talk to each other. To better understand what CORS is and why we use it, we'll briefly go over some background context.
+
+The [_origin_](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#definition_of_an_origin) of a piece of web content is that content's domain, protocol, and port. The [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) (SOP) is a security mechanism that restricts scripts on one origin from interacting with resources from another origin.
+
+```none
+# Same origin
+http://example.com:8080/ <==> http://example.com:8080/
+
+# Different origin (difference in domain, protocol, and port)
+http://example.com:8080/ =X= https://example1.com:8081/
+```
+
+However, as we all know, the Internet is an exciting place full of resources that can make websites better (e.g., importing images, extra fonts, making API calls, etc.).  Developers needed a new protocol to relax SOP and safely _share_ resources across different origins.
+
+Cross-Origin Resource Sharing is the mechanism that allows a web page to  _share_ resources across different origins. CORS provides an extra layer of protection by enabling servers and clients to define HTTP headers, which specify which external clients' scripts can access their resources.
+
+## Configuring CORS options for Apollo Server
+
+The batteries-included `apollo-server` [has a `cors` option](../api/apollo-server/#cors), which you can use to change your server's CORS behavior:
+
+```js
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["http://localhost:3000", "https://studio.apollographql.com"]
```

Comment:
Just below this codeblock, I'd add a single sentence that just says "The example above enables CORS requests from two origins blah blah"